src/Servers/Kestrel/Core/src/Internal/Http/Http1ChunkedEncodingMessageBody.cs

```diff
@@ -458,7 +458,8 @@ private void ParseChunkedSuffix(in ReadOnlySequence<byte> buffer, out SequencePo
             // Advance examined before possibly throwing, so we don't risk examining less than the previous call to ParseChunkedSuffix.
             examined = suffixBuffer.End;
 
-            if (suffixSpan[0] == '\r' && suffixSpan[1] == '\n')
+            // index 1 first, to elide the bound check for index 0
+            if (suffixSpan[1] == '\n' && suffixSpan[0] == '\r')
```

Comment:
I'm not sure if removing the few extra bounds checks is worth doing this backwards.

Reply:
Because it is unnatural?

Of course, the better solution would be if the JIT can recognize this (kind of flow-control) and elide the bound checks by himself without the need for some tricks.

The perf-win won't be huge, anyway. It safes one bound check (in this case, more in others). But the resulting code is smaller, and smaller code is almost always better.

I don't know either if it's (really) worth or not...(I would take it, as it's a tiny improment)

Reply:
Exactly. I don't doubt this is every-so-slightly faster, but I like keeping the code as natural as possible unless there's a significant perf win.

Reply:
So undo that change or close the PR as whole?

Reply:
I'm willing to listen to what others have to say, but my preference is to close this PR.

Reply:
Agree with @halter73 , the readability loss here is not worth the perf "gain". I'd imagine this wouldn't show up on a macro benchmark at all.